23_tiles_msleckman/2_process/src/read_in_reclassify.R

```diff
@@ -47,7 +47,23 @@ read_in_reclassify <- function(lc_tif_path, reclassify_legend,
   #return(reclassified_raster)
   
   }
-
+downsamp_cat <- function(raster, down_fact){
+  #' @param raster input raster
+  #' @param down_fact factor to downsample by in x and y directon
+  rast <- terra::rast(raster[[1]]) # convert to Spat
+  rast_seg <- terra::segregate(rast) # split categorical data
+  rast_down <- terra::aggregate(rast_seg, fact = down_fact,  sum) # downsample
+  
+  # convert to dataframe to plot wtih ggplot
+  ## find the mode for each aggregated cell
+  rast_down_df <- as.data.frame(rast_down, xy = TRUE) %>%
+    pivot_longer(!c(x,y)) %>%
+    filter(value > 0, name != 0) %>%
+    group_by(x,y) %>%
+    arrange(desc(value)) %>%
+    slice_max(1)
```

Comment:
Issue with slice here as it is grouped tibble and it doesn't recognize with col to apply slice_max() from. (at least that is my theory. 
I get no more error with this modification:  
```suggestion
      slice_max(value, n=1)
```